packages/grid/_modules_/grid/hooks/root/useColumns.ts

```diff
@@ -219,9 +218,10 @@ export function useColumns(
       const updatedCols: ColDef[] = [];
 
       const currentSortedCols = stateRef.current.all
-        .filter((c) => c.sortDirection != null)
-        .map((c) => ({ field: c.field, sort: c.sortDirection }));
-      if (isEqual(currentSortedCols, sortModel)) {
+        .filter((column) => column.sortDirection != null)
+        .map((column) => ({ field: column.field, sort: column.sortDirection }));
+
+      if (JSON.stringify(currentSortedCols) === JSON.stringify(sortModel)) {
```

Comment:
nahhh, can we have the isEqual function please. I use it all over. 
Lodash is treeshakable
Write your own...
JSON.stringify is bad!!! Do not use it for any other purpose than serialising to JSON. It's a hack to use it any other way.

```
JSON.stringify({ a: 1, b: 2}) !== JSON.stringify({ b: 2, a: 1}) 
```